packages/shorebird_cli/lib/src/shorebird_android_artifacts.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:path/path.dart' as p;
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+
+/// Thrown when multiple artifacts are found in the build directory.
+class MultipleArtifactsFoundException implements Exception {
+  MultipleArtifactsFoundException({
+    required this.buildDir,
+    required this.foundArtifacts,
+  });
+
+  final String buildDir;
+  final List<FileSystemEntity> foundArtifacts;
+
+  @override
+  String toString() {
+    return 'Multiple artifacts found in $buildDir: '
+        '${foundArtifacts.map((e) => e.path)}';
+  }
+}
+
+/// Thrown when no artifact is found in the build directory.
+class ArtifactNotFoundException implements Exception {
+  ArtifactNotFoundException({
+    required this.artifactName,
+    required this.buildDir,
+  });
+
+  final String artifactName;
+  final String buildDir;
+
+  @override
+  String toString() {
+    return 'Artifact $artifactName not found in $buildDir';
+  }
+}
+
+/// When building android artifacts, gradlew name these artifacts in an
+/// inconsistent way.
+///
+/// Example:
+///
+///  ## AABs
+///  Without flavors
+///  ...bundle/release/app-release.aab
+///
+///  With flavors
+///  ...bundle/flavor/app-flavor-release.aab
+///
+///  With multi dimensional flavors
+///  ...bundle/fullFlavorCamelCase/app-flavor1-flavor2-release.aab
+///
+/// The pattern follows for APKs.
+///
+/// The only thing that is consistent is the artifactId which is the name of the
+/// artifact will follow always the same order of name of the flavors.
+///
+/// To get around this, we create an identifier for the artifact
```

Comment:
Is there an upstream bug in Flutter tracking this? If so, can we link to it and add a TODO to revisit this workaround when the upstream bug is fixed?